packages/desktop-client/src/components/modals/EnvelopeIncomeBalanceMenuModal.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { type CSSProperties } from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { Menu } from '@actual-app/components/menu';
+import { styles } from '@actual-app/components/styles';
+import { Text } from '@actual-app/components/text';
+import { theme } from '@actual-app/components/theme';
+import { View } from '@actual-app/components/view';
+
+import {
+  BalanceWithCarryover,
+  CarryoverIndicator,
+} from '@desktop-client/components/budget/BalanceWithCarryover';
+import { useEnvelopeSheetValue } from '@desktop-client/components/budget/envelope/EnvelopeBudgetComponents';
+import {
+  Modal,
+  ModalCloseButton,
+  ModalHeader,
+  ModalTitle,
+} from '@desktop-client/components/common/Modal';
+import { CellValueText } from '@desktop-client/components/spreadsheet/CellValue';
+import { useCategory } from '@desktop-client/hooks/useCategory';
+import { type Modal as ModalType } from '@desktop-client/modals/modalsSlice';
+import { envelopeBudget } from '@desktop-client/queries/queries';
+
+type EnvelopeIncomeBalanceMenuModalProps = Omit<
+  Extract<ModalType, { name: 'envelope-income-balance-menu' }>['options'],
+  'month'
+>;
+
+export function EnvelopeIncomeBalanceMenuModal({
+  categoryId,
+  onCarryover,
+  onShowActivity,
+}: EnvelopeIncomeBalanceMenuModalProps) {
+  //const defaultMenuItemStyle: CSSProperties = {
+  //  ...styles.mobileMenuItem,
+  //  color: theme.menuItemText,
+  //  borderRadius: 0,
+  //  borderTop: `1px solid ${theme.pillBorder}`,
+  //};
+
+  const { t } = useTranslation();
+  const category = useCategory(categoryId);
+
+  const carryover = useEnvelopeSheetValue(
+    envelopeBudget.catCarryover(categoryId),
+  );
+
+  if (!category) {
+    return null;
+  }
+
+  return (
+    <Modal name="envelope-balance-menu">
```

Comment:
This might be the problem

```suggestion
    <Modal name="envelope-income-balance-menu">
```

Reply:
That was it.  The style is still not right, but much better.